### ISSUE TITLE
feat(transform): add orthogonal convolutional exponential transform adapted for jax/flowjax

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#5]
## Motivation

Introduce an *orthogonal convolutional exponential* (ConvExp) transform based on the implementation and theory from [ehoogeboom/convolution_exponential_and_sylvester](https://github.com/ehoogeboom/convolution_exponential_and_sylvester) for use with jax and flowjax. This transform allows for highly expressive, invertible convolutions, bridging orthogonality constraints with convolutional flows.

## Background

The convolution exponential builds on matrix exponentials:
- Matrix exponential is $e^M = \sum_{k=0}^\infty M^k/k!$
- For convolutions, this power series translates to repeated conv layers with weight sharing, accumulating terms.

**Theory key points:**
- Orthogonal convolution kernels ensure invertibility and stable Jacobian.
- Exponential of the convolution operator is efficiently computed as a truncated power series.
- The inverse is simple: negate the kernel.
- Log-determinant of the Jacobian is tractable via kernel diagonal.
- Spectral normalization keeps kernel norm bounded (prevents exploding/vanishing gradients).

## Original Implementation

Relevant functions from the reference repo ([convexp_module.py](https://github.com/ehoogeboom/convolution_exponential_and_sylvester/blob/main/models/transformations/convexp/convexp_module.py)):

```python
# Convolution exponential forward
for i in range(1, terms + 1):
    product = F.conv2d(product, kernel, padding=1, stride=(1, 1)) / i
    result = result + product
```

- Forward: accumulate conv2d powers (with $1/k!$ scaling).
- Inverse: apply same procedure, but with -kernel.
- Logdet: sum central diagonals of kernel.

## Deep Dive Notes

- **Scan/foriloop adaptation:**
    - Replace manual Python for-loop with `jax.lax.scan` or `foriloop` for high-performance, jit-compatible loop unrolling.
    - Loop maintains intermediate state and accumulates terms, enables truncation/early stopping.
    - Handles batch dimensions (B, C, H, W) properly; shape assertions are needed.
- **API design:**
    - Should mimic flowjax transform base class.
    - Parameters: kernel (trainable, spectral norm constraint), n_terms (training/eval), input_shape.
    - Methods: `forward`, `inverse`, `logdet`, optional 1x1 conv for extension.

## Proposed API for GaussFlows/Flowjax

```python
class OrthogonalConvExpTransform(flowjax.Transform):
    def __init__(self, kernel, n_terms=10, ...):
        # kernel shape: (C, C, K, K)
        # apply spectral normalization etc.
    def forward(self, x):
        # Apply conv_exp with scan/foriloop
    def inverse(self, x):
        # Apply inv_conv_exp (negate kernel)
    def logdet(self, x):
        # compute logdet via kernel diagonal (central pixel)
```

## Tasks

- [ ] Analyze and port core conv_exp logic to jax (with tests on shape, correctness).
- [ ] Implement forward/inverse using scan/foriloop.
- [ ] Add logdet computation for fast Jacobian determinant.
- [ ] Spectral normalization for kernel (use jax routines or adapt torch logic).
- [ ] Integrate with flowjax-style API: support initialization, batching, gradient flow.
- [ ] Document theory and practical considerations in docstring and module doc.
- [ ] Test and compare with reference implementation for correctness/invertibility.

## References
- [Repo](https://github.com/ehoogeboom/convolution_exponential_and_sylvester)
- [README: Theory, example code](https://github.com/ehoogeboom/convolution_exponential_and_sylvester/blob/main/README.md)
- [conv_exp function](https://github.com/ehoogeboom/convolution_exponential_and_sylvester/blob/main/models/transformations/convexp/functional.py#L42)
- SpectralNormConv implementation ([spectral.py](https://github.com/ehoogeboom/convolution_exponential_and_sylvester/blob/main/models/transformations/convexp/spectral.py))

---

**This transform has deep theoretical foundations and practical utility. Porting it to jax/flowjax will enable richer compositions with other flows in GaussFlows.**
